cmake/modules/AddSwift.cmake

```diff
@@ -2210,10 +2210,12 @@ function(_add_swift_executable_single name)
   # working around tests like Driver/linker.swift which copy/hard link Swift
   # executables to different directories without also copying the "libs"
   # directory. A more robust solution should be found.
+  # NOTE: Adding "${SWIFTLIB_DIR}/.." to the rpath was necessary to fix linker
+  # errors for Syntax/Parser tests on Linux.
   if("${SWIFTEXE_SINGLE_SDK}" STREQUAL "LINUX" AND NOT "${SWIFTEXE_SINGLE_SDK}" STREQUAL "ANDROID")
-    set(local_rpath "$ORIGIN:$ORIGIN/${swift_relative_library_path}:${SWIFTLIB_DIR}/linux:/usr/lib/swift/linux")
+    set(local_rpath "$ORIGIN:$ORIGIN/${swift_relative_library_path}:${SWIFTLIB_DIR}/linux:${SWIFTLIB_DIR}/..:/usr/lib/swift/linux")
```

Comment:
Note: there's a very small chance adding `$ORIGIN/${swift_relative_library_path}/..` may also be necessary for toolchain binaries to work.
I doubt it because `swift-syntax-parser-test` should not be installed into toolchains.

Reply:
This sort of fix implies that the problem exists in upstream. Is that so?